src/main/Sirius.java

```diff
@@ -0,0 +1,291 @@
+    public static final String UNMARK = "unmark";
+    public static final String TODO = "todo";
+    public static final String DEADLINE = "deadline";
+    public static final String EVENT = "event";
+    public static final String DELETE = "delete";
+
+
+    // some regexes
+    public static final String SPACE = " ";
+    public static final String EMPTY = "";
+    public static final String SLASH = "/";
+    public static final String STATUS_DELIMINATOR = "\\|";
+    public static final String SEPARATOR = "-----------------------------";
+
+    // data members
+    private static boolean isExit = true;
+    private static boolean isValidToProcess = true;
+    private static final ArrayList<Task> list = new ArrayList<>();
+
+    // some methods
+    public static void sayHello(){
+        System.out.println("""
+                -----------------------
+                Hello! I'm Sirius!
+                What can I do for you?
+                -----------------------
+                """);
+    }
+    public static void sayGoodbye(){
+        System.out.println("""
+                -----------------------
+                Bye! Hope to see you soon.
+                -----------------------
+                """);
+        isExit = false;
+    }
+    public static void markTask(String[] commandPieces, ArrayList<Task> list, boolean isMarked) {
+        System.out.println(SEPARATOR);
+        try {
+            int taskNumber = Integer.parseInt(commandPieces[1]);
+            if (isMarked) {
+                list.get(taskNumber - 1).setMarked(true);
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                list.get(taskNumber - 1).setMarked(false);
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+            System.out.println(list.get(taskNumber - 1).toString());
+            System.out.println(SEPARATOR);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("The task index is out of bounds! Please enter a valid task index");
+        } catch (NumberFormatException e) {
+            System.out.println("The task index must be a number! Please enter a valid index number!");
+        }
+    }
+    public static void addTask(String[] commandPieces, ArrayList<Task> list){
+        System.out.println(SEPARATOR);
+        String commandPrefix = commandPieces[0];
+        String taskName = commandPieces[1];
+        switch (commandPrefix) {
+            case DEADLINE:
+                list.add(new Deadline(taskName, false, commandPieces[2]));
+                break;
+            case EVENT:
+                list.add(new Event(taskName, false, commandPieces[2], commandPieces[3]));
+                break;
+            case TODO:
+                list.add(new Todo(taskName, false));
+            }
+        System.out.println("Got it. I've added this task:");
+        System.out.println(list.get(list.size()-1).toString());
```

Comment:
Consider adding a size attribute to the 'Task' class, as it is being used regularly and it is good practice to have all class related things together